File: bitmarket-common/src/main/java/com/bitwise/bitmarket/common/currency/CurrencyCode.java
package com.bitwise.bitmarket.common.currency;

public enum CurrencyCode {

    EUR("EUR"), USD("USD"), BTC("BTC");

    private final String currency;

    private CurrencyCode(final String currency) {
        this.currency = currency;
    }

    @Override
    public String toString() {
        return this.currency;
    }
}

File: coinffeine-common/src/main/scala/com/coinffeine/common/currency/CurrencyCode.scala
package com.coinffeine.common.currency

import java.util.Currency

@deprecated
class CurrencyCode(symbol: String) {

  val currency: Currency = Currency.getInstance(symbol)

  def apply(amount: BigDecimal): FiatAmount = FiatAmount(amount, currency)
}

object CurrencyCode {
  @deprecated val EUR = new CurrencyCode("EUR")
  @deprecated val USD = new CurrencyCode("USD")
}
